homeassistant/components/fitbit/__init__.py

```diff
@@ -29,6 +29,10 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     )
     try:
         await fitbit_api.async_get_access_token()
+    except aiohttp.ClientResponseError as err:
+        if 400 <= err.status < 500:
```

Comment:
Are we sure there are no 400-range errors where a re-authentication won't help?

Reply:
You're right, this is overly broad and not for a specific reason (copied from another integration which may have a more specific reason for its error code matching choices or maybe should be revisited). 
 I looked at the docs for the API and it is very specific about which error codes indicate the refresh token is expired so i've reduced the scope here and added tests for other error codes to just retry

More on their spec here https://dev.fitbit.com/build/reference/web-api/developer-guide/application-design/#Access-tokens

> When an access token expires, an HTTP 401 error will be returned. Your application will need to use the refresh token to obtain a new access token and refresh token pair.